packages/core/utils/src/event-bus/message-aggregator.ts

```diff
@@ -28,11 +29,25 @@ export class MessageAggregator implements IMessageAggregator {
 
   saveRawMessageData<T>(
     messageData:
-      | EventBusTypes.MessageFormat<T>
-      | EventBusTypes.MessageFormat<T>[],
-    options?: Record<string, unknown>
+      | EventBusTypes.RawMessageFormat<T>
+      | EventBusTypes.RawMessageFormat<T>[],
+    {
+      options,
+      sharedContext,
+    }: { options?: Record<string, unknown>; sharedContext?: Context } = {}
   ): void {
-    this.save(buildEventMessages(messageData, options))
+    const messages = Array.isArray(messageData) ? messageData : [messageData]
```

Comment:
Why did we move this out of the compose? I guess we'll have to write the same thing over and over again

Reply:
I ve removed the old one in favor of the new one, but they was doing the exact same thing, if you look at the function at the end we call the same as well with the composed message